### DISCUSSION TITLE
what-is-ebpf/ebpf-in-cloud-native-environments/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/cloud-native-library#25] [Announcements]
# what-is-ebpf/ebpf-in-cloud-native-environments/

近年来，云原生应用已呈指数级增长。在本章中，我将讨论为什么 eBPF 如此适合于云原生环境。为了更具象化，我将提到 Kubernetes，但同样适用于任何容器平台。 每台主机一个内核 要理解为什么 eBPF 在云原生世界中如此

https://lib.jimmysong.io/what-is-ebpf/ebpf-in-cloud-native-environments/

**Comment by rootsongjc** [comment]
简单地说，作者的观点是 eBPF 开启了上帝视角，可以看到主机上所有的活动，而 sidecar 只能观测到 pod 内的活动，只要搞好进程隔离，基于 eBPF 的 proxy per-node 才是最佳选择。
